examples/webgl_buffergeometry.html

```diff
@@ -51,105 +57,133 @@
-					const cz = z + Math.random() * d - d2;
+						const cx = x + Math.random() * d - d2;
+						const cy = y + Math.random() * d - d2;
+						const cz = z + Math.random() * d - d2;
 
-					positions.push( ax, ay, az );
-					positions.push( bx, by, bz );
-					positions.push( cx, cy, cz );
+						positions.push( ax, ay, az );
+						positions.push( bx, by, bz );
+						positions.push( cx, cy, cz );
 
-					// flat face normals
+						// flat face normals
 
-					pA.set( ax, ay, az );
-					pB.set( bx, by, bz );
-					pC.set( cx, cy, cz );
+						pA.set( ax, ay, az );
+						pB.set( bx, by, bz );
+						pC.set( cx, cy, cz );
 
-					cb.subVectors( pC, pB );
-					ab.subVectors( pA, pB );
-					cb.cross( ab );
+						cb.subVectors( pC, pB );
+						ab.subVectors( pA, pB );
+						cb.cross( ab );
 
-					cb.normalize();
+						cb.normalize();
 
-					const nx = cb.x;
-					const ny = cb.y;
-					const nz = cb.z;
+						const nx = cb.x;
+						const ny = cb.y;
+						const nz = cb.z;
 
-					normals.push( nx, ny, nz );
-					normals.push( nx, ny, nz );
-					normals.push( nx, ny, nz );
+						normals.push( nx, ny, nz );
+						normals.push( nx, ny, nz );
+						normals.push( nx, ny, nz );
 
-					// colors
+						// colors
 
-					const vx = ( x / n ) + 0.5;
-					const vy = ( y / n ) + 0.5;
-					const vz = ( z / n ) + 0.5;
+						const vx = ( x / n ) + 0.5;
+						const vy = ( y / n ) + 0.5;
+						const vz = ( z / n ) + 0.5;
 
-					color.setRGB( vx, vy, vz );
+						color.setRGB( vx, vy, vz );
 
-					colors.push( color.r, color.g, color.b );
-					colors.push( color.r, color.g, color.b );
-					colors.push( color.r, color.g, color.b );
+						if ( data.VertexAlpha ) {
 
-				}
+							const alpha = Math.random();
+							colors.push( color.r, color.g, color.b, alpha );
+							colors.push( color.r, color.g, color.b, alpha );
+							colors.push( color.r, color.g, color.b, alpha );
+
+						} else {
+
+							colors.push( color.r, color.g, color.b );
+							colors.push( color.r, color.g, color.b );
+							colors.push( color.r, color.g, color.b );
+
+						}
+
+					}
 
-				function disposeArray() {
+					function disposeArray() {
 
-					this.array = null;
+						this.array = null;
+
+					}
+
+					geometry.setAttribute( 'position', new THREE.Float32BufferAttribute( positions, 3 ).onUpload( disposeArray ) );
+					geometry.setAttribute( 'normal', new THREE.Float32BufferAttribute( normals, 3 ).onUpload( disposeArray ) );
+					let numColorChannels = data.VertexAlpha ? 4 : 3;
+					geometry.setAttribute( 'color', new THREE.Float32BufferAttribute( colors, numColorChannels ).onUpload( disposeArray ) );
+
+					geometry.computeBoundingSphere();
+
+					const material = new THREE.MeshPhongMaterial( {
+						color: 0xaaaaaa, specular: 0xffffff, shininess: 250,
+						side: THREE.DoubleSide, vertexColors: true,
+						vertexAlpha: data.VertexAlpha, transparent: data.VertexAlpha
+					} );
+
+					scene.remove( mesh );
```

Comment:
You should not call `Scene.remove()` when `mesh` is `undefiend`.

I suggest you remove the usage of dat.gui and thus the option to switch between vertex alpha. Just create the data like that right away.

Reply:
Sure. I also updated the screenshot.